test/test_python_code.py

```diff
@@ -142,6 +147,33 @@ def test_get_image(self, xcffib_test):
             output_format, root, 0, 0, width, height, plane_mask).reply()
         reply.data.buf()
 
+    def test_ge_generic_event(self, xcffib_test):
+
+        def make_ge_generic(extension, event_type, length):
+            buf = six.BytesIO()
+            buf.write(struct.pack("=B", 35))
+            buf.write(struct.pack("=B2xIH22xI", extension, length, event_type, 0))
+            buf_len = len(buf.getvalue())
+            print(buf_len)
+            if buf_len < 32:
+                buf.write(struct.pack("x" * (32 - buf_len)))
+            return buf.getvalue()
+
+        setup = xcffib_test.conn.get_setup()
+        screen = setup.roots[0]
+        root = screen.root
+
+        for l in range(1000):
+            try:
+                xcffib_test.xproto.SendEvent(False, root, EventMask.NoEvent, make_ge_generic(10, 20, l))
+                xcffib_test.conn.flush()
+
+                e = xcffib_test.conn.wait_for_event()
+            except:
+                continue
+
+        assert isinstance(e, xcffib.xproto.GeGenericEvent)
```

Comment:
Maybe also send some data through and verify that the offsets are where you expect them?

Reply:
This doesn't work at all at the moment!